### PR TITLE
feat: [CI-6007]: Support for github action step

### PR DESCRIPTION
**Pull request by shubham149**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/39504)

### CODE REVIEW

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmActionStepSerializer.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+@Singleton
+public class VmActionStepSerializer {
+  @Inject CIExecutionServiceConfig ciExecutionServiceConfig;
+
+  public VmRunStep serialize(ActionStepInfo actionStepInfo, String identifier) {
+    String uses =
+        RunTimeInputHandler.resolveStringParameter("Uses", "Action", identifier, actionStepInfo.getUses(), true);
+    Map<String, String> with = resolveMapParameter("with", "Action", identifier, actionStepInfo.getWith(), false);
+
+    Map<String, String> env = resolveMapParameter("with", "Action", identifier, actionStepInfo.getEnv(), false);
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmActionStepSerializer.java:26]: Field name is env

**Review by shubham149 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39504)* status: 26 file(s) to review. 2. unresolved discussions. 0 of 1 approvals obtained. Waiting on team(s): sto. 

___
*[`332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeStepUtils.java` line 310 at r1](https://reviewable.io/reviews/harness/harness-core/39504#-NFs0UJA6IFxqXZkRHDn-r1-310:-NGGHlvF9CAskn5sba7C:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/4d661517be2a65b92348d46aa2ad00ed7659f089/332-ci-manager/service/src/main/java/io/harness/ci/execution/integrationstage/K8InitializeStepUtils.java#L310)):*
<details><summary><i>Previously, harshjain12 (Harsh Jain) wrote…</i></summary><blockquote>

 Step is only applicable for builds on VM infrastructure
</blockquote></details>

Done.
___
*[`332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmActionStepSerializer.java` line 26 at r1](https://reviewable.io/reviews/harness/harness-core/39504#-NFs0UJA6IFxqXZkRHDr-r1-26:-NGGHmcJ3RSfXfvzycEA:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/4d661517be2a65b92348d46aa2ad00ed7659f089/332-ci-manager/service/src/main/java/io/harness/ci/execution/serializer/vm/VmActionStepSerializer.java#L26)):*
<details><summary><i>Previously, harshjain12 (Harsh Jain) wrote…</i></summary><blockquote>

Field name is env
</blockquote></details>

Done.

**Comment by harshjain12** [comment]
@shubham149 let us also add check in save pipeline call.

**Review by harshjain12 — APPROVED** [review]
> Reviewed 17 of 26 files at r1, 7 of 8 files at r2, 9 of 9 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39504)* status: Waiting on team(s): sto.

**Review by harshjain12 — APPROVED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/39504)* status: Waiting on team(s): sto.

**Pull request merged**